chapters/equations.tex

```diff
@@ -632,12 +632,16 @@ \section{Events and Synchronization}\label{events-and-synchronization}
 \end{nonnormative}
 
 \begin{example}
+Two different errors caused by non-discrete-time expressions:
 \begin{lstlisting}[language=modelica]
-Limit1 = noEvent(x1 > 1);  // Error since Limit1 is a discrete-time variable
-when noEvent(x1>1) or x2>10 then // error, when-conditions is not a discrete-time expression
-  Close = true;
+when noEvent(x1 > 1) or x2 > 10 then // When-condition must be discrete-time
+  close = true;
 end when;
+above1 = noEvent(x1 > 1);            // above1 is discrete-time variable
 \end{lstlisting}
+The when-condition error is a direct violation of the requirement that the when-condition must be discrete-time.
+The error in the equation with \lstinline!above1! is only indirect; since \lstinline!above1! requires a dicsrete-time solution, it cannot be solved from this equation.
```

Comment:
The `above1` issue is not indirect, since it is a non-Real equation  and https://specification.modelica.org/master/operators-and-expressions.html#discrete-time-expressions require that both sides must be discrete-time expressions in that case. (Clearly it is based on the fact that it would be singular.)

Reply:
Thus the previous text would work.

Reply:
I wouldn't say that the previous text would work, since it doesn't give any hint that the rule in effect has to do with neither side of the equation being `Real`.  I had forgotten about that rule (probably because I find the rule redundant given the other requirements we have for discrete-valued variables).  To help readers like me, couldn't we at least replace the text I added by an explanation with reference to the rule?

Reply:
Yes, we can, of course, add a reference to the rule.

Reply:
How about that?